src/storage/v2/indices/vector_index.cpp

```diff
@@ -133,142 +136,203 @@ unum::usearch::metric_kind_t VectorIndex::MetricFromName(std::string_view name)
   if (name == "sorensen") {
     return unum::usearch::metric_kind_t::sorensen_k;
   }
+  if (name == "jaccard") {
+    return unum::usearch::metric_kind_t::jaccard_k;
+  }
   throw query::VectorSearchException(
       fmt::format("Unsupported metric name: {}. Supported metrics are l2sq, ip, cos, haversine, divergence, pearson, "
                   "hamming, tanimoto, and sorensen.",
                   name));
 }
 
-bool VectorIndex::CreateIndex(const VectorIndexSpec &spec, utils::SkipList<Vertex>::Accessor &vertices,
-                              std::optional<SnapshotObserverInfo> const &snapshot_info) {
-  try {
-    // Create the index
-    const unum::usearch::metric_punned_t metric(spec.dimension, spec.metric_kind, unum::usearch::scalar_kind_t::f32_k);
+utils::BasicResult<VectorIndexStorageError, void> VectorIndex::CreateIndex(
+    const VectorIndexSpec &spec, utils::SkipList<Vertex>::Accessor &vertices,
+    std::optional<SnapshotObserverInfo> const &snapshot_info) {
+  const auto label_prop = LabelPropKey{spec.label, spec.property};
+  // Create the index
+  const unum::usearch::metric_punned_t metric(spec.dimension, spec.metric_kind, unum::usearch::scalar_kind_t::f32_k);
+
+  // use the number of workers as the number of possible concurrent index operations
+  const unum::usearch::index_limits_t limits(spec.capacity, FLAGS_bolt_num_workers);
+
+  if (pimpl->index_name_to_label_prop_.find(spec.index_name) != pimpl->index_name_to_label_prop_.end()) {
+    return VectorIndexStorageError::VectorIndexAlreadyExists;
+  }
```

Comment:
would be better if you would use std::set instead of vector and then rely on try_emplace whether it was inserted. In this way you have to add enum InconsistentState which hides what is actually going on